### PR TITLE
Add docs for setting up WSL 2 & Docker

### PR DESCRIPTION
**Pull request by spowelljr**
Related https://github.com/kubernetes/minikube/issues/13293

Adding documentation for users wanting to setup WSL 2 and Docker Desktop for Windows.

### CODE REVIEW

File: site/content/en/docs/drivers/docker.md [hunk 1]
```diff
@@ -9,7 +9,46 @@ aliases:
...
+```shell
+dockerd-rootless-setuptool.sh install -f
+docker context use rootless
+
+minikube start --driver=docker --container-runtime=containerd
+```
+
+The `--container-runtime` flag must be set to "containerd" or "cri-o".
+
+The restrictions of rootless `kind` apply to minikube with rootless docker as well.
```
Comment (medyagh) [inline site/content/en/docs/drivers/docker.md:49]: this reference could be confusing the user, to go form one cli site to another, then they might follow kind commands and expect it to work

Comment (spowelljr) [reply site/content/en/docs/drivers/docker.md:49]: I didn't add this, I just copied it from another file, do you want the two lines removed?

Comment (spowelljr) [reply site/content/en/docs/drivers/docker.md]: Removed

**Comment by klaases** [comment]
Hi @medyagh, do you want to re-review this PR based on Steven's changes?

**Comment by medyagh** [comment]
excellent ! I like it

**Pull request merged**